theseus/src/api/pack.rs

```diff
@@ -62,20 +62,45 @@ impl From<String> for PackFileHash {
 
 #[derive(Serialize, Deserialize, Eq, PartialEq, Hash)]
 #[serde(rename_all = "camelCase")]
-enum EnvType {
+pub enum EnvType {
     Client,
     Server,
 }
 
 #[derive(Serialize, Deserialize, Clone, Hash, PartialEq, Eq)]
 #[serde(rename_all = "kebab-case")]
-enum PackDependency {
+pub enum PackDependency {
     Forge,
     FabricLoader,
     QuiltLoader,
     Minecraft,
 }
 
+// the modloader version to use, set to "latest", "stable", or the ID of your chosen loader. defaults to latest
+// This may differ from a String, because it considers forge version IDs which may include a hyphen separating two versions
+// This only considers the last part of the version string, after the hyphen, in these cases
+#[derive(Serialize, Eq, PartialEq, Debug, Clone)]
+#[serde(transparent)]
+pub struct LoaderVersionString(pub String);
+impl<'de> Deserialize<'de> for LoaderVersionString {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+
+        // if there's a hyphen, only take the part after the hyphen, if there's no hyphen, take the whole thing
+        Ok(LoaderVersionString::from(s))
+    }
+}
+// if there's a hyphen, only take the part after the hyphen, if there's no hyphen, take the whole thing
+// This is designed with Forge versions in mind, such as 1.20-1.2.2 (the minecraft version is 1.20, the forge version is 1.2.2)
+impl From<String> for LoaderVersionString {
+    fn from(s: String) -> Self {
+        LoaderVersionString(s.split('-').last().unwrap_or(&s).to_string())
+    }
+}
+
```

Comment:
does this have to be done through serde? feels weird to do it this way. It should only be done on export IMO with some handling there.